test/src/problems/FailingProblem.C

```diff
@@ -17,21 +17,24 @@ InputParameters
 FailingProblem::validParams()
 {
   InputParameters params = FEProblem::validParams();
-  params.addRequiredParam<unsigned int>("fail_step", "The timestep to fail");
+  params.addRequiredParam<std::vector<int>>("fail_steps", "The timestep(s) to fail");
   return params;
 }
 
 FailingProblem::FailingProblem(const InputParameters & params)
-  : FEProblem(params), _failed(false), _fail_step(getParam<unsigned int>("fail_step"))
+  : FEProblem(params), _fail_steps(getParam<std::vector<int>>("fail_steps"))
 {
 }
 
 bool
 FailingProblem::converged()
 {
-  if (!_failed && (_t_step == static_cast<int>(_fail_step)))
+  if (_t_step == _fail_steps.front())
   {
-    _failed = true;
+    if (_fail_steps.size() > 1)
+      _fail_steps.erase(_fail_steps.begin());
+    else
+      _fail_steps = {-1};
```

Comment:
why -1? empty should be ok?

Reply:
also wont the erase work for size 1 as well?

Reply:
re: -1, it needs to have a dummy value for the comparison against the timestep to work on line 32, and -1 will never be accidentally hit over the course of the program as you obviously won't be going back in time over a timestep

re:erase, again we want to keep one element in the vector for the comparison. 

could probably change the process logic if really necessary for elegance's sake to avoid either

Reply:
with a find operation you can use an empty vector